### PR TITLE
Updated the code to take dynamic input from user

### PR DESCRIPTION
**Pull request by ymaheshwari1**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description
Updated code to get inputs from the user with informative messages.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#1211

### CODE REVIEW

**Comment by ymaheshwari1** [comment]
@atarax665 Please review.

File: Java/search/BinarySearch.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}
+/*
+Input:
+Enter the length of the array: 5
+Enter array values: 2 3 4 5 6
+Enter the value whose index needs to be searched: 4
+
+Output:
+Index of 4 is: 3
+
```
Comment (atarax665) [inline Java/search/BinarySearch.java]: Add space and time complexity of Binary search.

File: Java/search/LinearSearch.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+		for(int i=0; i<length; i++) {
+			arr[i] = sc.nextInt();
+		}
+		System.out.print("Enter the value whose index needs to be searched: ");
+        int value = sc.nextInt();
+        System.out.println("Index of "+ value +" is: "+linSearch(arr,value,length));
+    }
+}
+/*
+Input:
```
Comment (atarax665) [inline Java/search/LinearSearch.java]: Add time and space complexity.

**Review by atarax665 — CHANGES_REQUESTED** [review]
Format your code according to Java style guide. You can use https://www.techiedelight.com/tools/java

**Comment by atarax665** [comment]
@ymaheshwari1  You don't need to delete and push a new file. You can just edit the existing ones. Do check the other files in the folder too, if they have static input edit them too.

File: Java/search/BinarySearch.java (hunk 1 above)
Comment (ymaheshwari1) [reply Java/search/BinarySearch.java]: I have added the complexities of the algorithm.

File: Java/search/LinearSearch.java (hunk 2 above)
Comment (ymaheshwari1) [reply Java/search/LinearSearch.java]: I have added the complexities of the algorithm.

File: Java/search/Binary_search.java [hunk 3]
```diff
@@ -1,29 +1,49 @@
...
+	//binary search algorithm explanation in its readme file in wiki section
+	//this can performed recursively also
+	//The average and worst case time complexity is O(log n)
+	//The space complexity is O(1).
+	static int binSearch(int[] a, int l, int h, int key) {
+		while (l<= h) {
+			int mid = (l + h) / 2;
+			if (key == a[mid])
+				return mid + 1; //index starts from zero so position has to be incremented by one.
+			else if (key<a[mid])
```
Comment (atarax665) [inline Java/search/Binary_search.java]:
```suggestion
			else if (key < a[mid])
```

File: Java/search/Binary_search.java [hunk 4]
```diff
@@ -1,29 +1,49 @@
...
+		return -1;
+	}
+	public static void main(String[] args) {
+		//to perform binary search the array must be in sorted order
+		Scanner sc = new Scanner(System.in);
+		System.out.print("Enter the length of the array: ");
+		int length = sc.nextInt();
+		int arr[] = new int[length];
+		System.out.print("Enter array values: ");
+		for (int i = 0; i<length; i++) {
```
Comment (atarax665) [inline Java/search/Binary_search.java]:
```suggestion
		for (int i = 0; i < length; i++) {
```

File: Java/search/Linear_search.java [hunk 5]
```diff
@@ -1,19 +1,35 @@
...
-    public static void main(String[] args){
-        int[] a={2,7,8,9,1,5,3,4};
-        int key=8;
-        System.out.println(linsearch(a,key,8));
-    }
+	//It iterates through the array to find key.
+	//The average and worst case time complexity O(n).
+	//The worst case space complexity is O(1).
+	static int linSearch(int[] a, int k, int length) {
+		for (int i = 0; i<length; i++) {
```
Comment (atarax665) [inline Java/search/Linear_search.java]:
```suggestion
		for (int i = 0; i < length; i++) {
```

File: Java/search/Linear_search.java [hunk 6]
```diff
@@ -1,19 +1,35 @@
...
+		}
+		return -1; //returns -1 if not found
+	}
+	public static void main(String[] args) {
+		Scanner sc = new Scanner(System.in);
+		System.out.print("Enter the length of the array: ");
+		int length = sc.nextInt();
+		int arr[] = new int[length];
+		System.out.print("Enter array values: ");
+		for (int i = 0; i<length; i++) {
```
Comment (atarax665) [inline Java/search/Linear_search.java]:
```suggestion
		for (int i = 0; i < length; i++) {
```

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Review by atarax665 — APPROVED** [review]
LGTM :)

**Review by vatsalkesarwani12 — APPROVED** [review]
👍🏻

**Pull request merged**